### DISCUSSION TITLE
Setting `dirtyFields` based on differences between `values` & `dirtyValues`

### DISCUSSION THREAD

**Discussion by denchen** [react-hook-form/react-hook-form#9745] [Q&A] [answered]
Is there a way to have the `dirtyFields` be set to  mark any field as dirty if the value in `values` is different from `dirtyValues`?

My use case is a multi-page form, where each page has its own instance of `useForm()`. All the values for the entire form is stored in React Context. If the user decides to go back to a previous page, of course, the `useForm()` for that page is reinitialized, but this time with the current form `values` stored in Context. I want those fields to be already marked as dirty.

**Comment by leapful** [comment, marked as answer]
@denchen 

You can use `useEffect` hook to reset form values with your new value in React context with option `keepDefaultValues`, just remember to set default values on `useForm` hook in order to dirty check between default values and your current value from context.

Working demo: https://codesandbox.io/s/eager-currying-rp8goz?file=/src/App.tsx

**Reply by denchen** [reply]
Thanks! That worked perfectly.
